docs/experimental-features.md

```diff
@@ -218,36 +218,86 @@ configured, the daemon will fail to start.
 ---
 
 ## ipfs p2p
-Allows to tunnel TCP connections through Libp2p streams
+
+Allows tunneling of TCP connections through Libp2p streams. If you've ever used
+port forwarding with SSH (the `-L` option in openssh), this feature is quite
+similar.
 
 ### State
+
 Experimental
 
 ### In Version
+
 master, 0.4.10
 
 ### How to enable
-P2P command needs to be enabled in config
 
-`ipfs config --json Experimental.Libp2pStreamMounting true`
+The `p2p` command needs to be enabled in config:
+
+```sh
+> ipfs config --json Experimental.Libp2pStreamMounting true
+```
 
 ### How to use
 
-Basic usage:
+First, pick a protocol name for your application. Think of the protocol name as
+a port number, just significantly more user-friendly. In this example, we're
+going to use `/kickass/1.0`.
+
+**Setup:**
+
+1. A "server" node with peer ID `$SERVER_ID`
+2. A "client" node.
+
+**On the "server" node:**
+
+First, start your application and have it listen on `$APP_PORT`.
+
+Then, configure the p2p listener by running:
+
+```sh
+> ipfs p2p forward /kickass/1.0 /ipfs /ip4/127.0.0.1/tcp/$APP_PORT
```

Comment:
whats the /ipfs in the middle for?

Reply:
It's the listen address. I could also make it accept the node peerid. I just wanted one command to handle forwarding in both directions.

It would be cleaner if we could specify this as a multiaddress. We'd need to define a multiaddr proto for string protocol names, probably something like `/mux/protoName`. With that the command wouldn't have to accept magic strings, and the protocol name could get integrated with the address. Creating a libp2p service would look like

```
ipfs p2p /mux/someProto /ip4/127.0.0.1/tcp/$APP_PORT
```

And forwarding local connections to a service:
```
ipfs p2p /ip4/127.0.0.1/tcp/$APP_PORT /ipfs/QmPeer/mux/someProto
```

Does that make more sense?

Reply:
Yeah, that makes sense... Its just a bit confusing. It would be nice to be able to make that simpler somehow. @Stebalien @Kubuxu @schomatis thoughts?

Reply:
I'm getting the error `Error: protocol name must be within '/p2p/' namespace` while trying to replicate the example (to better understand this conversation as I don't really know much about libp2p).

Reply:
Ok, I get it now, changing `/kickass/1.0` to `/p2p/kickass/1.0` fixes that and the netcat example is working.

Reply:
oh, should update that too

Reply:
So, going back to the original question, what seems confusing to me (again, I know nothing about libp2p so feel free to ignore this comment) is to have the p2p port in the front, which seems as if the forwarding is happening only at the p2p layer when in fact the message is routed through different layers (p2p-TCP).

If there were only two arguments (as suggested by @magik6k's solution that for me would be the ideal case) then the from-to relationship would become more evident, but if that isn't possible at the moment I would rather have two different commands that would clearly indicate to the user when the message is received at the p2p layer and routed to the TCP layer (and when the other way around); or at least I would reorganize the command arguments to group the p2p protocol with the p2p address.